### ISSUE TITLE
[FEATURE] Set max detection points, dets per image, and data annotation instances per image constants in config

### ISSUE THREAD

**Issue by MichaelMonashev** [rwightman/efficientdet-pytorch#120]
How to set number of detections?

**Comment by MichaelMonashev** [comment]
I have to change MAX_DETECTIONS_PER_IMAGE in YOUR code to predict all objects. How to increase this limit without your code patching?

**Comment by rwightman** [comment]
There are module level constants defined for those limits right now, they are currently set at comfortable values for COCO, VOC, OI, etc

They can be changed on the anchor side and dataset side 
https://github.com/rwightman/efficientdet-pytorch/blob/master/effdet/anchors.py#L46
https://github.com/rwightman/efficientdet-pytorch/blob/master/effdet/data/loader.py#L12

Low priority TODO to move them into config.

**Comment by Ekta246** [comment]
Does this means if I have only 1 class, 1 object per image i.e, 1 bounding box, I should substitute the MAX_NUM_INSTANCES=1 in the dataloader?

**Comment by MichaelMonashev** [comment]
> Does this means if I have only 1 class, 1 object per image i.e, 1 bounding box, I should substitute the MAX_NUM_INSTANCES=1 in the dataloader?

No.

**Comment by Ekta246** [comment]
The changing of the dets_per image and max_num_instances varies the validation loss drastically. I am still finding the reason of the overfitting and validation loss divergence.

**Comment by MichaelMonashev** [comment]
Thank you, @rwightman .
